src/state/routing/utils.ts

```diff
@@ -41,8 +42,15 @@ export function computeRoutes(
       }
 
       return {
```

Comment:
This requires a lot of looping that isn't necessary. Could you improve it by doing something like this?

    getRouteProtocol(route: (V3PoolInRoute | V2PoolInRoute)[]): Protocol {
      if (route.every((pool) => pool.type === 'v2-pool')) return Protocol.V2
      if (route.every((pool) => pool.type === 'v3-pool')) return Protocol.V3
      return Protocol.MIXED
    }

You could optimize it more but this seems sufficient. The old `isV2Route`-style functions were type guards, but the `genericPoolPairParser` makes it so that you no longer need to cast to a type before parsing the route.

Reply:
Yeah this is a lot more elegant. This also solves the comment above about the mixed routes not exactly being comprehensive